proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -726,6 +748,27 @@ def eq_equal_flow_vol_interface_io(b, t, x):
             return prop_interface_io.flow_vol_phase['Liq'] ==\
                    prop_io.flow_vol_phase['Liq']
 
+        # constraints for additional variables (i.e. variables not used in other constraints)
+        @self.Constraint(self.flowsheet().config.time)
+        def eq_recovery_vol_phase(b, t):
+            return (b.recovery_vol_phase[t, 'Liq'] ==
+                    b.properties_permeate[t].flow_vol_phase['Liq'] /
+                    b.feed_side.properties_in[t].flow_vol_phase['Liq'])
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.config.property_package.component_list)
+        def eq_recovery_mass_phase_comp(b, t, j):
+            return (b.recovery_mass_phase_comp[t, 'Liq', j] ==
+                    b.properties_permeate[t].flow_mass_phase_comp['Liq', j] /
+                    b.feed_side.properties_in[t].flow_mass_phase_comp['Liq', j])
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.solute_list)
+        def eq_rejection_phase_comp(b, t, j):
+            return (b.rejection_phase_comp[t, 'Liq', j] ==
+                    1 - (b.properties_permeate[t].conc_mass_phase_comp['Liq', j] /
+                         b.feed_side.properties_in[t].conc_mass_phase_comp['Liq', j]))
```

Comment:
Just reconsidering whether we should add intrinsic rejection as well, which would be the same constraint, except you'd divide by the feed concentration at the membrane-interface. Experimentalists make use of intrinsic solute rejection, and the value gives a sense of the "real" rejection of a membrane. 

I think it would be nice to have observed_rejection and intrinsic_rejection.

Reply:
I looked into this and intrinsic rejection is tricky and I'd like to know more about the use case before we add it. The observed rejection is 1 - concentration in permeate/concentration in feed, for the intrinsic what is the denominator? The concentration at the membrane interface at the inlet or at the outlet? 

Essentially, the disconnect is the observed rejection is a module metric while the intrinsic rejection is a local metric (i.e. at a specific point)

I'm leaving this off for now, but we can always add it.

Reply:
This [paper by Kim & Hoek ](https://doi.org/10.1016/j.desal.2005.05.017) gives an example of how intrinsic rejection can be used (check out equations 6-7 for a quick glance). Essentially, intrinsic rejection is the "real" rejection of the membrane, similar to how the membrane-interface osmotic pressure is the "real" osmotic pressure that must be overcome. I suppose it'd be preferable to make this a local variable, so perhaps we can reconsider in another PR (could also get the advisory board's take on use cases).